### PR TITLE
Migration of Semi-supervision and domain adaptation with AdaMatch to keras 3…

### PR DESCRIPTION
**Pull request by maitry63**
This PR migrates the Semi-supervision and domain adaptation with AdaMatch tutorial from Keras 2 to Keras 3. The implementation is now fully backend-agnostic, allowing it to run seamlessly on JAX, PyTorch, and TensorFlow.

Colab file - [Notebook](https://colab.sandbox.google.com/gist/maitry63/54effe25a1795f2ee424c717c2f9a78a/adamatch.ipynb#scrollTo=qllcMkUJ31If)

### CODE REVIEW

**Comment by maitry63** [comment]
@gemini-code-assist review

**Comment by maitry63** [comment]
@gemini-code-assist review

**Review by sachinprasadhs — APPROVED** [review]
LGTM

**Comment by sachinprasadhs** [comment]
Just one comment. 
Previously, all the Conv2D had    
```
     kernel_initializer=INIT,
        kernel_regularizer=regularizers.l2(WEIGHT_DECAY),
```
Now only selective `Conv2D` layers has it, why is such inconsistency?
Can you please clarify it or address it?

**Comment by maitry63** [comment]
> Just one comment. Previously, all the Conv2D had
> 
> ```
>      kernel_initializer=INIT,
>         kernel_regularizer=regularizers.l2(WEIGHT_DECAY),
> ```
> 
> Now only selective `Conv2D` layers has it, why is such inconsistency? Can you please clarify it or address it?

Thanks for pointing this out. I have  updated the implementation to ensure consistency across all Conv2D layers. Now all convolution layers now uniformly use the specified weight initialization and L2 weight decay regularization.
